tool/sync_default_gems.rb

```diff
@@ -598,52 +549,73 @@ def resolve_conflicts(gem, sha, edit)
         end
         if editor
           system([editor, conflict].join(' '))
+          return system(*%w"git add --", *conflict)
         end
       end
+      return false
     end
 
-    # Attempt to commit the cherry-pick
-    system({"GIT_EDITOR"=>"true"}, *%W"git cherry-pick --no-edit --continue") || nil
+    return true
   end
 
-  def remove_toplevel_addtions(gem, sha)
-    # Forcibly remove any new top-level entries, and any changes under
-    # /test/fixtures, /test/lib, or /tool.
-    changed = pipe_readlines(%W"git diff --name-only -z HEAD~..HEAD --")
-    toplevels = changed.map {|f| f[%r[\A(?!tool/)[^/]+/?]]}.compact
-    toplevels.delete_if do |top|
-      if system(*%w"git checkout -f HEAD~ --", top, err: File::NULL)
-        # previously existent path
-        system(*%w"git checkout -f HEAD --", top, out: File::NULL)
+  def filter_pickup_files(changed, ignore_file_pattern, base)
+    toplevels = {}
+    remove = []
+    ignore = []
+    changed = changed.reject do |f|
+      case
+      when toplevels.fetch(top = f[%r[\A[^/]+(?=/|\z)]m]) {
+             remove << top unless
+               toplevels[top] = system(*%w"git cat-file -e", "#{base}:#{top}", err: File::NULL)
+           }
+        # Remove any new top-level directories.
         true
+      when !f.include?("/"),
+           f.start_with?("test/fixtures/", "test/lib/", "tool/")
+        # Forcibly reset any top-level entries, and any changes under
+        # /test/fixtures, /test/lib, or /tool.
+        ignore << f
+      when ignore_file_pattern.match?(f)
+        # Forcibly reset any changes matching ignore_file_pattern.
+        ignore << f
       end
     end
-    unless toplevels.empty?
-      puts "Remove files added to toplevel: #{toplevels.join(', ')}"
-      system(*%w"git rm -r --", *toplevels)
+    return changed, remove, ignore
+  end
+
+  def pickup_files(gem, changed, picked)
+    # Forcibly remove any files that we don't want to copy to this
+    # repository.
+
+    ignore_file_pattern = ignore_file_pattern_for(gem)
+
+    base = picked ? "HEAD~" : "HEAD"
+    changed, remove, ignore = filter_pickup_files(changed, ignore_file_pattern, base)
+
+    unless remove.empty?
+      puts "Remove added files: #{remove.join(', ')}"
+      system(*%w"git rm -fr --", *remove)
+      if picked
+        system(*%w"git commit --amend --no-edit --", *remove, %i[out err] => File::NULL)
+      end
     end
-    tools = changed.select {|f|f.start_with?("test/fixtures/", "test/lib/", "tool/")}
-    unless tools.empty?
-      system(*%W"git rm -r --", *tools)
```

Comment:
Why did you remove this `git rm`? This doesn't seem to work (See #alerts-sync on Slack for logs).